ejb3/src/main/java/org/jboss/as/ejb3/tx/CMTTxInterceptor.java

```diff
@@ -243,7 +246,13 @@ protected Object invokeInOurTx(InterceptorContext invocation, final EJBComponent
                 } catch (EJBException | RemoteException e) {
                     throw e;
                 } catch (RuntimeException e) {
-                    throw ae != null ? e : new EJBException(e);
+                    if(ae != null && !ae.isRollback()) {
+                        except = e;
```

Comment:
do we need to wrap it as EJBException(e)?

Reply:
In case of the tx fail an EJBTransactionRolledbackException will be thrown which is an EJBException. (https://github.com/jboss-set/eap-additional-testsuite/blob/master/modules/src/main/java/org/jboss/additional/testsuite/jdkall/present/ejb/exceptions/ClientExceptionTestCase.java#L50). I think in the other cases it should remain a RuntimeException.

Reply:
if it is `EJBTransactionRolledbackException` and hence `EJBException`, it will already been handled by the `catch (EJBException | RemoteException)` above this clause.  So here it will be some non-EJBExcpetion RuntimeException.

Reply:
The RuntimeException at this point is the Exception that is thrown by MyApplicationExceptionNoRollbackRuntimeException (https://github.com/jboss-set/eap-additional-testsuite/blob/master/modules/src/main/java/org/jboss/additional/testsuite/jdkall/present/ejb/exceptions/MyApplicationExceptionNoRollbackRuntimeException.java#L12).
The EJBTransactionRolledbackException is thrown because of the RuntimeException here : https://github.com/jboss-set/eap-additional-testsuite/blob/master/modules/src/main/java/org/jboss/additional/testsuite/jdkall/present/ejb/exceptions/HelloBean.java#L41